tests/tests_firewall_fact.yml

```diff
@@ -0,0 +1,86 @@
+- name: Test firewall ansible fact
+  hosts: all
+  become: true
+  tasks:
+    - name: Test firewall ansible fact
+      block:
+        - name: Start with default configuration
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+              - previous: replaced
+
+        - name: Get default ansible fact
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+
+        - name: Fail if firewall_config is undefined
+          fail:
+            msg: Ansible fact firewall_config is not defined
+          when: firewall_config is undefined
+
+        - name: Fail if firewall_config custom has attributes
+          vars:
+            custom_item: "{{ firewall_config.custom[item] | list }}"
+          fail:
+            msg: "{{ custom_item }}"
+          when: custom_item | length != 0
+          loop: "{{ firewall_config.custom.keys() | list }}"
```

Comment:
I think you could just check for the length of custom:
```
        - name: Fail if firewall_config custom has attributes
          fail:
            msg: There should be no custom config
          when: firewall_config.custom | length > 0
```
This assumes that `firewall_config.custom` is always defined (I think it is).  The `length` of a `dict` is the number of keys, so an empty `dict` should have `length == 0`

Reply:
Yes firewall_config.custom is always defined, but so are all of the options, so in any case the length of firewall_config.custom would be 5 or 6

Reply:
ok - if you make the changes I suggested in the python code, the emptiness check should be easier, but you'll have to check if `when: "'custom' in firewall_config"` or similar